src/policy/marksweepspace/native_ms/block_list.rs

```diff
@@ -164,14 +164,21 @@ impl BlockList {
         BlockListIterator { cursor: self.first }
     }
 
-    /// Sweep all the blocks in the block list.
-    pub fn sweep_blocks<VM: VMBinding>(&self, space: &super::MarkSweepSpace<VM>) {
+    /// Release unmarked blocks, and sweep other blocks in the block list. Used by eager sweeping.
+    pub fn release_and_sweep_blocks<VM: VMBinding>(&self, space: &super::MarkSweepSpace<VM>) {
```

Comment:
In `release_and_sweep_block`, `sweep` is called if the block `attempt_release` returns `false`.  But if a block is unallocated, `attempt_release` will still return `false`, and this function will still call `sweep`.  But it doesn't make sense to sweep an unallocated block.

I think it doesn't even make sense to call `attempt_release` on an unallocated block in the first place.  If we guarantee that we never add any unallocated block into any `BlockList`, we can just let `attempt_release` panic when it sees a block is unallocated.

It may be necessary to be a bit sloppy about block states if the race between `SweepChunk` and `BlockList::release_and_sweep_blocks` still exists because the block state may suddenly change.  Now that we removed that race, we may simply assume the block cannot be unallocated before we call `attempt_release`.